crates/bevy_render/src/render_resource/pipeline_specializer.rs

```diff
@@ -18,11 +18,29 @@ use core::{fmt::Debug, hash::Hash};
 use thiserror::Error;
 use tracing::error;
 
+/// A trait that allows constructing different variants of a render pipeline from a key.
+///
+/// Note: If your key is a ZST, you do not need to "specialize" anything. Just create the render
```

Comment:
I don't really get what this note is saying 🤔

Reply:
I would maybe just rephrase it to say that specialization is intended to modify the pipeline descriptor and cached on the basis of the key. The relevant scenario here would be if someone was using `AsBindGroup` derive without the #[bind_group_data] attr, then you don't need to specialize.